src/GovUk.Education.ExploreEducationStatistics.Admin.Tests/Controllers/Api/ReleasesControllerTests.cs

```diff
@@ -448,6 +448,72 @@ public async Task CreateReleaseAmendment()
             result.AssertOkResult(amendmentCreatedResponse);
         }
 
+        [Fact]
+        public async Task UploadBulkZipDataSetsToTempStorage()
+        {
+            // Arrange
+            var dataSetFiles = new List<ArchiveDataSetFileViewModel>();
+
+            var releaseDataFileService = new Mock<IReleaseDataFileService>(Strict);
+
+            releaseDataFileService
+                .Setup(s => s.ValidateAndUploadBulkZip(
+                    It.IsAny<Guid>(),
+                    It.IsAny<IFormFile>(),
+                    default))
+                .ReturnsAsync(dataSetFiles);
+
+            var controller = BuildController(releaseDataFileService: releaseDataFileService.Object);
+
+            // Act
+            var result = await controller.UploadBulkZipDataSetsToTempStorage(
+                Guid.NewGuid(),
+                MockFile("bulk.zip"),
+                default);
+
+            // Assert
+            VerifyAllMocks(releaseDataFileService);
+
+            result.AssertOkResult(dataSetFiles);
+        }
+
+        [Fact]
+        public async Task ImportBulkZipDataSetsFromTempStorage()
+        {
+            // Arrange
+            var dataFileInfo = new List<DataFileInfo>
+            {
+                new() { FileName = "one.csv", Name = "Data set title", Size = "1024" },
+            };
+
+            var importRequests = new List<ArchiveDataSetFileViewModel>
+            {
+                new(){ DataFileId = Guid.NewGuid(), MetaFileId = Guid.NewGuid(), Title = "Data set title", DataFilename = "one.csv", MetaFilename = "one.meta.csv", DataFileSize = 1024, MetaFileSize = 128 }
+            };
+
+            var releaseDataFileService = new Mock<IReleaseDataFileService>(Strict);
+
+            releaseDataFileService
+                .Setup(s => s.SaveDataSetsFromTemporaryBlobStorage(
+                    It.IsAny<Guid>(),
+                    It.IsAny<List<ArchiveDataSetFileViewModel>>(),
+                    default))
+                .ReturnsAsync(dataFileInfo);
+
+            var controller = BuildController(releaseDataFileService: releaseDataFileService.Object);
+
+            // Act
+            var result = await controller.ImportBulkZipDataSetsFromTempStorage(
+                Guid.NewGuid(),
+                importRequests,
+                default);
+
+            // Assert
+            VerifyAllMocks(releaseDataFileService);
+
+            result.AssertOkResult(dataFileInfo);
+        }
+
```

Comment:
I could remove them, but I was just following the existing pattern in this class. Would you agree that they provide at least _some_ benefit, and removing them would be a detriment, albeit minor?

Reply:
I agree they provide some benefit, yes, so leave them in if you want. They check the service call is actually made and that service response is returned and that the input/output is not modified.

I don't think anyone would mind though if you were to go in and delete `ReleasesControllerTests.cs` entirely!

The following failure test cases in that class test a little bit more i.e. that different failure return types from the service layer are handled as bad requests / not found, etc, but this should be generic behaviour that's tested once elsewhere.

```
DeleteDataFiles_Fail_UnableToFindMetaFileToDelete
DeleteReleaseVersion_Returns_NotFound
DeleteReleaseVersion_Returns_ValidationProblem
```

I said before in the previous comment that the handling of success/failure is already unit tested but I might be wrong about that - I was expecting to see that `EitherTaskExtensions` is covered by unit tests but it looks like it might not be.